src/core/blob.cpp

```diff
@@ -20,31 +26,37 @@ blob::blob(session & s)
 
 blob::~blob()
 {
-    delete backEnd_;
+    if (backEnd_)
+        delete backEnd_;
 }
 
 std::size_t blob::get_len()
 {
+    assert(backEnd_);
```

Comment:
al right, this is true that in release mode this is not checked. I will replace them with exceptions